lib/streamlit/state/session_state.py

```diff
@@ -278,6 +304,62 @@ def _missing_key_error_message(key: str) -> str:
     )
 
 
+class WidgetStateReport(Protocol[T_co]):
```

Comment:
How about `RegisterWidgetResult` as a type name?

Reply:
I like it better than `...Report`, so changing the name to that. But it does come with some [misleading connotations](https://doc.rust-lang.org/std/result/) for me. That is a minor issue, though.